### ISSUE TITLE
`torch.lu_solve` gives an unintuitive error message when the inputs are not batched

### ISSUE THREAD

**Issue by bamos** [pytorch/pytorch#22243]
## 🐛 Bug

`torch.lu_solve` gives an unintuitive error message when the inputs are not batched

## To Reproduce

```python
torch.manual_seed(0)
n = 4
L = torch.randn(n,n)
A = L.mm(L.t())
b = torch.randn(n)
b.lu_solve(*torch.lu(A))
```

Output:

```
RuntimeError: expected non-empty 3D tensor, got size: [4, 4]
```

## Expected behavior

Manually batching:
```python
b.unsqueeze(0).lu_solve(*torch.lu(A.unsqueeze(0)))
```

Output:

```
tensor([[-1.7040,  1.0375, -5.7181, -4.8389]])
```

**Comment by vishwakftw** [comment]
Thank you for reporting this. I’m working on adding batching to lu_solve, currently hindered by a MAGMA issue.

**Comment by vishwakftw** [comment]
Closed by 0ab19d66ee4e9b4fd4cb854c1dcffe65f6b92464.
